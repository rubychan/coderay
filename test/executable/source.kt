class Test {
    val text1: String = "abc \' def \u0000 \n \" hehe "
    val text2: List<Char> = listOf('a', '\n', '\'', '"', '\"')
    val numbers = listOf(0, 12, 1.0f, 1.0, 1L, 0x1f, -1, -12, -1.0f, -1.0, -1L)

    val template = "abc${1 + "b"}def"
    val template2 = "abc${1 + 'b'}def"
    val template3 = "abc $var def"

    val multiline = """ first line $var ${1 + 1}
        second line
        and quotes: ' " ''  ""  ok
        """

    val innerBraces = " before ${ if (true) { 1 } else { 2 } }"

    var v: Int = 0

    fun function(): ReturnType {
    }

    fun <T : Any> parametrizedFunction(): T = TODO()

    fun references() {
        super.references()
        this.references()
    }

    @Annotation
    class Annotated

    inner class Inner<in T, out E>

    object O

    companion object {
    }
}