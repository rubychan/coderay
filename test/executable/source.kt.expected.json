[{"type":"text","text":"class","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"Test","kind":"class"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"{","kind":"operator"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"text1","kind":"ident"},{"type":"text","text":":","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"String","kind":"predefined_type"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"text","text":"abc ","kind":"content"},{"type":"text","text":"\\'","kind":"char"},{"type":"text","text":" def ","kind":"content"},{"type":"text","text":"\\u0000","kind":"char"},{"type":"text","text":" ","kind":"content"},{"type":"text","text":"\\n","kind":"char"},{"type":"text","text":" ","kind":"content"},{"type":"text","text":"\\\"","kind":"char"},{"type":"text","text":" hehe ","kind":"content"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"text2","kind":"ident"},{"type":"text","text":":","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"List","kind":"predefined_type"},{"type":"text","text":"<","kind":"operator"},{"type":"text","text":"Char","kind":"type"},{"type":"text","text":">","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"listOf","kind":"ident"},{"type":"text","text":"(","kind":"operator"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"'","kind":"delimiter"},{"type":"text","text":"a","kind":"content"},{"type":"text","text":"'","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"'","kind":"delimiter"},{"type":"text","text":"\\n","kind":"content"},{"type":"text","text":"'","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"'","kind":"delimiter"},{"type":"text","text":"\\'","kind":"char"},{"type":"text","text":"'","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"'","kind":"delimiter"},{"type":"text","text":"\"","kind":"content"},{"type":"text","text":"'","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"'","kind":"delimiter"},{"type":"text","text":"\\\"","kind":"content"},{"type":"text","text":"'","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"numbers","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"listOf","kind":"ident"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":"0","kind":"integer"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"12","kind":"integer"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"1.0f","kind":"float"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"1.0","kind":"float"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"1L","kind":"integer"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"0x1f","kind":"hex"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"-","kind":"operator"},{"type":"text","text":"1","kind":"integer"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"-","kind":"operator"},{"type":"text","text":"12","kind":"integer"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"-","kind":"operator"},{"type":"text","text":"1.0f","kind":"float"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"-","kind":"operator"},{"type":"text","text":"1.0","kind":"float"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"-","kind":"operator"},{"type":"text","text":"1L","kind":"integer"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"template","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"text","text":"abc","kind":"content"},{"type":"text","text":"${","kind":"operator"},{"type":"block","action":"open","kind":"initial"},{"type":"text","text":"1","kind":"integer"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"+","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"text","text":"b","kind":"content"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"}","kind":"operator"},{"type":"block","action":"close","kind":"initial"},{"type":"text","text":"def","kind":"content"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"template2","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"text","text":"abc","kind":"content"},{"type":"text","text":"${","kind":"operator"},{"type":"block","action":"open","kind":"initial"},{"type":"text","text":"1","kind":"integer"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"+","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"'","kind":"delimiter"},{"type":"text","text":"b","kind":"content"},{"type":"text","text":"'","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"}","kind":"operator"},{"type":"block","action":"close","kind":"initial"},{"type":"text","text":"def","kind":"content"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"template3","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"text","text":"abc ","kind":"content"},{"type":"text","text":"$var","kind":"ident"},{"type":"text","text":" def","kind":"content"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"multiline","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"\"\"\"","kind":"delimiter"},{"type":"text","text":" first line ","kind":"content"},{"type":"text","text":"$var","kind":"ident"},{"type":"text","text":" ","kind":"content"},{"type":"text","text":"${","kind":"operator"},{"type":"block","action":"open","kind":"initial"},{"type":"text","text":"1","kind":"integer"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"+","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"1","kind":"integer"},{"type":"text","text":"}","kind":"operator"},{"type":"block","action":"close","kind":"initial"},{"type":"text","text":"\n        second line\n        and quotes: ' ","kind":"content"},{"type":"text","text":"\"","kind":"content"},{"type":"text","text":" ''  ","kind":"content"},{"type":"text","text":"\"","kind":"content"},{"type":"text","text":"\"","kind":"content"},{"type":"text","text":"  ok\n        ","kind":"content"},{"type":"text","text":"\"\"\"","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"val","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"innerBraces","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"block","action":"open","kind":"string"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"text","text":" before ","kind":"content"},{"type":"text","text":"${","kind":"operator"},{"type":"block","action":"open","kind":"initial"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"if","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":"true","kind":"predefined_constant"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"{","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"1","kind":"integer"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"}","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"else","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"{","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"2","kind":"integer"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"}","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"}","kind":"operator"},{"type":"block","action":"close","kind":"initial"},{"type":"text","text":"\"","kind":"delimiter"},{"type":"block","action":"close","kind":"string"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"var","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"v","kind":"ident"},{"type":"text","text":":","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"Int","kind":"type"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"0","kind":"integer"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"fun","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"function","kind":"ident"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":":","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"ReturnType","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"{","kind":"operator"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"}","kind":"operator"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"fun","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"<","kind":"operator"},{"type":"text","text":"T","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":":","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"Any","kind":"type"},{"type":"text","text":">","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"parametrizedFunction","kind":"ident"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":":","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"T","kind":"ident"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"=","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"TODO","kind":"comment"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"fun","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"references","kind":"ident"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"{","kind":"operator"},{"type":"text","text":"\n        ","kind":"space"},{"type":"text","text":"super","kind":"local_variable"},{"type":"text","text":".","kind":"operator"},{"type":"text","text":"references","kind":"ident"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":"\n        ","kind":"space"},{"type":"text","text":"this","kind":"local_variable"},{"type":"text","text":".","kind":"operator"},{"type":"text","text":"references","kind":"ident"},{"type":"text","text":"(","kind":"operator"},{"type":"text","text":")","kind":"operator"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"}","kind":"operator"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"@Annotation","kind":"annotation"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"class","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"Annotated","kind":"class"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"inner","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"class","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"Inner","kind":"class"},{"type":"text","text":"<","kind":"operator"},{"type":"text","text":"in","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"T","kind":"ident"},{"type":"text","text":",","kind":"operator"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"out","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"E","kind":"ident"},{"type":"text","text":">","kind":"operator"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"object","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"O","kind":"ident"},{"type":"text","text":"\n\n    ","kind":"space"},{"type":"text","text":"companion","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"object","kind":"keyword"},{"type":"text","text":" ","kind":"space"},{"type":"text","text":"{","kind":"operator"},{"type":"text","text":"\n    ","kind":"space"},{"type":"text","text":"}","kind":"operator"},{"type":"text","text":"\n","kind":"space"},{"type":"text","text":"}","kind":"operator"}]
